tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassloaderTestRunner.java

```diff
@@ -0,0 +1,199 @@
+        }
+        else
+        {
+            delegate = new BlockJUnit4ClassRunner(isolatedTestClass);
+        }
+
+        injectExtensionPluginsClassLoaders(classLoaderTestRunner, isolatedTestClass);
+    }
+
+    private Class<?> getTestClass(Class<?> clazz) throws InitializationError
+    {
+        try
+        {
+            return classLoaderTestRunner.loadClassWithApplicationClassLoader(clazz.getName());
+        }
+        catch (Exception e)
+        {
+            throw new InitializationError(e);
+        }
+    }
+
+    /**
+     * Creates the {@link ClassLoaderTestRunner} with the isolated class loaders.
+     *
+     * @param klass
+     * @return creates a {@link ClassLoaderTestRunner} that would be used to run the test. This way the test will be isolated and it will behave
+     * similar as an application running in a Mule standalone container.
+     */
+    private static ClassLoaderTestRunner createClassLoaderTestRunner(Class<?> klass)
+    {
+        // Initializes utility classes
+        ClassPathURLsProvider classPathURLsProvider = new DefaultClassPathURLsProvider();
+        MavenDependenciesResolver mavenDependenciesResolver = new DependencyGraphMavenDependenciesResolver();
+        MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping = new MuleMavenMultiModuleArtifactMapping();
+        MuleClassLoaderRunnerFactory classLoaderRunnerFactory = new MuleClassLoaderRunnerFactory();
+        ClassPathClassifier classPathClassifier = new MuleClassPathClassifier();
+
+        // Does the classification and creation of the isolated ClassLoader
+        ArtifactUrlClassification artifactUrlClassification = classPathClassifier.classify(new DefaultClassPathClassifierContext(klass, classPathURLsProvider.getURLs(),
+                                                                                           mavenDependenciesResolver.buildDependencies(), mavenMultiModuleArtifactMapping));
+        return classLoaderRunnerFactory.createClassLoader(klass, artifactUrlClassification);
+    }
+
+    /**
+     * Invokes the method to inject the plugin/extension class loaders for registering the extensions to the {@link org.mule.runtime.core.api.MuleContext}.
+     *
+     * @param classLoaderTestRunner
+     * @param isolatedTestClass
+     * @throws Throwable
+     */
+    private static void injectExtensionPluginsClassLoaders(ClassLoaderTestRunner classLoaderTestRunner, Class<?> isolatedTestClass) throws Throwable
+    {
+        TestClass testClass = new TestClass(isolatedTestClass);
+        Class<? extends Annotation> artifactContextAwareAnn = (Class<? extends Annotation>) classLoaderTestRunner.loadClassWithApplicationClassLoader(PluginClassLoadersAware.class.getName());
+        List<FrameworkMethod> contextAwareMethods = testClass.getAnnotatedMethods(artifactContextAwareAnn);
+        for (FrameworkMethod method : contextAwareMethods)
```

Comment:
Should not be required to have a PluginClassLoadersAware annotated method? I mean, the test wonk work properly otherwise. Moving a step forward... do we need to support other test class different from the ArtifactFunctionalTestCase?